js/auth.jsx

```diff
@@ -36,35 +35,38 @@ var auth = {
   },
 
   getCurrentUser: function(cb) {
-    function getCurrentUserLocal(cb) {
+    // Attempt to get the current user locally and check the server if that fails
+    var currentUser = getCurrentUserLocal();
+    if (!currentUser) {
+      return getCurrentUserRemote(cb);
+    } else {
+      cb(currentUser);
+    }
+
+    function getCurrentUserLocal() {
       if (typeof window.localStorage === 'undefined') {
-        return false;
-      }
-      var currentUser = JSON.parse(window.localStorage.getItem('currentUser'));
-      if (!currentUser) {
-        return false;
-      } else {
-        cb(currentUser);
-        return true;
+        return null;
       }
+      return JSON.parse(window.localStorage.getItem('currentUser'));
     }
 
     function getCurrentUserRemote(cb) {
```

Comment:
Make sure to do the callBack on null if no user is logged in. JSON will send a null user back as "", which caused problems.